src/state/selectors/selectors.ts

```diff
@@ -210,7 +211,20 @@ export const combineTooltipPayload = (
     const finalNameKey: DataKey<any> | undefined = settings?.nameKey; // ?? tooltipAxis?.nameKey;
     let tooltipPayload: unknown;
     if (tooltipAxis?.dataKey && !tooltipAxis?.allowDuplicatedCategory && Array.isArray(sliced)) {
-      tooltipPayload = findEntryInArray(sliced, tooltipAxis.dataKey, activeLabel);
+      if (activeLabel || !sliced?.length) {
+        tooltipPayload = findEntryInArray(
+          sliced,
+          entry => {
+            if (!Array.isArray(entry)) {
+              return get(entry, tooltipAxis.dataKey as string);
+            }
+            return entry.find((item: any) => item.dataKey === tooltipAxis.dataKey)?.value;
+          },
+          activeLabel,
+        );
+      } else {
+        tooltipPayload = tooltipPayloadSearcher(sliced, activeIndex, computedData, finalNameKey);
```

Comment:
wouldn't this do the same thing as below and behave as if `allowDuplicatedCategory` was true?